packages/pilot/src/pages/Settlements/SettlementsPage.js

```diff
@@ -0,0 +1,260 @@
+  translate('translations'),
+  connect(
+    mapStateToProps
+  )
+)
+
+const buildInitialState = (query = {}) => {
+  const initialState = {
+    count: parseInt(query?.count, 10) || 15,
+    offset: parseInt(query?.offset, 10) || 1,
+    ...query,
+  }
+  if ((!query.payment_date_start || !query.payment_date_end)
+    && !query.liquidation_arrangement_id) {
+    const today = moment()
+    const start = moment().subtract(7, 'days')
+    return {
+      ...initialState,
+      payment_date_end: today.toISOString(),
+      payment_date_start: start.toISOString(),
+    }
+  }
+
+  return initialState
+}
+
+const SettlementsPage = ({
+  client,
+  hasExternalContract,
+  recipientId,
+  t,
+}) => {
+  const history = useHistory()
+  const columns = getDefaultColumns({ t })
+  const location = useLocation()
+  const [isLastPaged, setIsLastPaged] = useState(false)
+  const parsed = qs.parse(location.search.substring(1))
+  const [filters, setFilters] = useState(buildInitialState(parsed))
+  const [rows, setRows] = useState([])
+  const [loading, setLoading] = useState(false)
+
+  const onFilterChange = useCallback((key, value) => {
+    setFilters(v => ({
+      ...v,
+      [key]: value || undefined,
+    }))
+  }, [])
+
+  const onConfirm = useCallback(async () => {
+    const parsedFilters = {
+      ...filters,
+    }
+    if (parsedFilters.liquidation_arrangement_id) {
+      delete parsedFilters.payment_date_end
+      delete parsedFilters.payment_date_start
+    }
+    const filtersString = qs.stringify(parsedFilters)
+    history.push({
+      search: filtersString,
+    })
+
+    try {
+      setLoading(true)
+      const {
+        last_paged: lastPaged,
+        settlements,
+      } = await client.settlements.all({
+        recipientId,
+        ...qs.parse(filtersString),
+      })
+      setIsLastPaged(lastPaged)
+      setRows(settlements)
+    } catch (e) {
+      setRows([])
+    } finally {
+      setLoading(false)
+    }
+  }, [client, filters, history, recipientId])
+
+  useEffect(() => {
+    onConfirm()
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const onReset = useCallback(() => {
+    setFilters(buildInitialState({}))
+    onConfirm()
+  }, [onConfirm])
+
+  const onPageChange = useCallback((offset) => {
+    onFilterChange('offset', parseInt(offset, 10))
+  }, [onFilterChange])
+
+  const onCountChange = useCallback((e) => {
+    onFilterChange('count', parseInt(e.target.value, 10))
+  }, [onFilterChange])
+
+  if (!hasExternalContract) {
+    history.push('/')
+  }
+
+  const paginationComponent = (
+    <Pagination
+      currentPage={filters.offset}
+      disabled={isLastPaged || loading}
+      onPageChange={onPageChange}
+      totalPages={filters.offset}
+    />
+  )
+
+  return (
+    <Grid>
+      <Row>
+        <Col
+          desk={12}
+          palm={12}
+          tablet={12}
+          tv={12}
+        >
+          <Filter
+            dateLimit={moment().add('days', 7)}
```

Comment:
```suggestion
            dateLimit={moment().add(7, 'days').endOf('day')}
```

Inverti a ordem dos argumentos, pq na forma que tá, é depracated no moment. E também adicionei o endOf('day') pra garantir que no último dia, vai funcionar até o final do dia. Tava ficando vermelho no último dia.